omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitsConfigurationController2_0Test.java

```diff
@@ -0,0 +1,99 @@
+import org.openmrs.module.webservices.rest.web.v1_0.controller.RestControllerTestUtils;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.util.OpenmrsConstants;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class VisitsConfigurationController2_0Test extends RestControllerTestUtils {
+
+	private AdministrationService administrationService;
+
+	private VisitService visitService;
+
+	@Before
+	public void before() throws SchedulerException {
+		administrationService = Context.getAdministrationService();
+		visitService = Context.getVisitService();
+	}
+
+	@Test
+	public void shouldGetCurrentConfiguration() throws Exception {
+		// set initial configuration
+
+		// enable visits
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, "true");
+		administrationService.saveGlobalProperty(enableVisitsGp);
+
+		// set encounter visit handler
+		GlobalProperty visitEncounterHandlerGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER,
+				"org.openmrs.api.handler.NoVisitAssignmentHandler");
+		administrationService.saveGlobalProperty(visitEncounterHandlerGp);
+
+		// set visit types to auto close
+		VisitType testVisitType = visitService.getVisitTypeByUuid(RestTestConstants1_9.VISIT_TYPE_UUID);
+		GlobalProperty visitTypesToAutoCloseGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_TYPES_TO_AUTO_CLOSE,
+				testVisitType.getName());
+		administrationService.saveGlobalProperty(visitTypesToAutoCloseGp);
+
+		// make GET call
+		MockHttpServletRequest req = request(RequestMethod.GET, "visitconfiguration");
+		SimpleObject result = deserialize(handle(req));
+
+		// assert response
+		assertNotNull(result);
+		assertEquals(true, PropertyUtils.getProperty(result, "enableVisits"));
+		assertEquals("org.openmrs.api.handler.NoVisitAssignmentHandler",
+				PropertyUtils.getProperty(result, "encounterVisitsAssignmentHandler"));
+
+		List<Object> visitTypesToAutoClose = (List<Object>) PropertyUtils.getProperty(result, "visitTypesToAutoClose");
+		assertEquals(1, visitTypesToAutoClose.size());
+	}
+
+	@Test
+	public void shouldUpdateCurrentConfiguration() throws Exception {
+		// update configuration
+		String json =
+				"{\"enableVisits\": true,\"encounterVisitsAssignmentHandler\": \"org.openmrs.api.handler.NoVisitAssignmentHandler\",\"startAutoCloseVisitsTask\": true,\"visitTypesToAutoClose\":[{\"uuid\":\""
+						+ RestTestConstants1_9.VISIT_TYPE_UUID + "\"}]}";
+		handle(newPostRequest("visitconfiguration", json));
+
+		// make GET call
+		MockHttpServletRequest req = request(RequestMethod.GET, "visitconfiguration");
+		SimpleObject result = deserialize(handle(req));
+
+		// assert response
+		assertNotNull(result);
```

Comment:
How do we tell that these were not so before the REST call?

Reply:
Resolved here: https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/a63dca7d7ddd4582c609e226e096ba4bd1c77717